Bottom: 57e0de0a85d88d3a29c050879e1efff780e734d2
Top:    e38edb9a0acc6cfa5f5dbdc45d89aac2bf5729c0
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-07-04 21:36:48 +0800

Refresh of exp

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 3af7374..c67db25 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1590,6 +1590,8 @@ static void tun_rx_batched(struct tun_struct *tun, struct tun_file *tfile,
 	if (rcv) {
 		struct sk_buff *nskb, *tmp;
 
+		list_add_tail(&skb->list, &list);
+
 		local_bh_disable();
 		list_for_each_entry_safe(nskb, tmp, &list, list) {
 			list_del_init(&nskb->list);
